src/ImageSharp/Processing/Processors/Transforms/NonAffineTransformProcessor.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Helpers;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    /// <summary>
+    /// Provides the base methods to perform non-affine transforms on an image.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class NonAffineTransformProcessor<TPixel> : InterpolatedTransformProcessorBase<TPixel>
```

Comment:
I'd rather call this `ProjectiveTransformProcessor` or `HomographyTransformProcessor`

Reply:
Either work for me. I'll let you decide.